applications/serial_lte_modem/src/slm_settings.c

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <logging/log.h>
+#include <string.h>
+#include <zephyr.h>
+#include <settings/settings.h>
+#include <drivers/uart.h>
+#include "slm_at_fota.h"
+
+LOG_MODULE_REGISTER(slm_config, CONFIG_SLM_LOG_LEVEL);
+
+/**
+ * Serial LTE Modem setting page for persistent data
+ */
+uint8_t fota_type;		/* FOTA: image type */
+uint8_t fota_stage;		/* FOTA: stage of FOTA process */
+uint8_t fota_status;		/* FOTA: OK/Error status */
+int32_t fota_info;		/* FOTA: failure cause in case of error or download percentage*/
+
+bool uart_configured;		/* UART: first-time configured */
+struct uart_config slm_uart;	/* UART: config */
+
+static int settings_set(const char *name, size_t len, settings_read_cb read_cb, void *cb_arg)
+{
+	if (!strcmp(name, "fota_type")) {
+		if (len != sizeof(fota_type))
+			return -EINVAL;
+		if (read_cb(cb_arg, &fota_type, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "fota_stage")) {
+		if (len != sizeof(fota_stage))
+			return -EINVAL;
+		if (read_cb(cb_arg, &fota_stage, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "fota_status")) {
+		if (len != sizeof(fota_status))
+			return -EINVAL;
+		if (read_cb(cb_arg, &fota_status, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "fota_info")) {
+		if (len != sizeof(fota_info))
+			return -EINVAL;
+		if (read_cb(cb_arg, &fota_info, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "uart_configured")) {
+		if (len != sizeof(uart_configured))
+			return -EINVAL;
+		if (read_cb(cb_arg, &uart_configured, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "uart_baudrate")) {
+		if (len != sizeof(slm_uart.baudrate))
+			return -EINVAL;
+		if (read_cb(cb_arg, &slm_uart.baudrate, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "uart_parity")) {
+		if (len != sizeof(slm_uart.parity))
+			return -EINVAL;
+		if (read_cb(cb_arg, &slm_uart.parity, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "uart_stop_bits")) {
+		if (len != sizeof(slm_uart.stop_bits))
+			return -EINVAL;
+		if (read_cb(cb_arg, &slm_uart.stop_bits, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "uart_data_bits")) {
+		if (len != sizeof(slm_uart.data_bits))
+			return -EINVAL;
+		if (read_cb(cb_arg, &slm_uart.data_bits, len) > 0)
+			return 0;
+	} else if (!strcmp(name, "uart_flow_ctrl")) {
+		if (len != sizeof(slm_uart.flow_ctrl))
+			return -EINVAL;
+		if (read_cb(cb_arg, &slm_uart.flow_ctrl, len) > 0)
+			return 0;
+	}
+
+	return -ENOENT;
+}
+
+int slm_settings_init(void)
+{
+	int ret;
+	struct settings_handler slm_settings_conf = {
```

Comment:
Shouldn't this be somewhere in the static RAM instead of the stack? The `settings_register()` doesn't seem to copy it's contents, just appends a node to the list.

Reply:
I'll move it to .BSS then

Reply:
Sorry for the delay, I somehow overlooked this PR. What happened to the `slm_settings.c`, it seems no longer present in the commit.